src/main/java/beam/physsim/jdeqsim/cacc/sim/Road.java

```diff
@@ -224,127 +177,50 @@ private double  getNextAvailableTimeForLeavingStreet(double simTime){
-             * long, it takes time for the gap to come back.
-             *
-             * As long as interestedInEnteringRoad is not empty, newly generated
-             * gaps get used by the new cars (see leaveRoad)
-             */
-            if (gap_ == null) {
-                gap_ = new LinkedList<>();
-            } else {
-                gap_.clear();
-            }
-
-            interestedInEnteringRoad_.add(vehicle);
-
-            /*
-             * the first car interested in entering a road has to wait
-             * 'stuckTime' the car behind has to wait an additional stuckTime
-             * (this logic was adapted to adhere to the C++ implementation)
-             */
-            double nextStuckTime=0;
-
-            if (deadlockPreventionMessages_.size() > 0) {
-                nextStuckTime=deadlockPreventionMessages_.getLast().getMessageArrivalTime() + config.getSqueezeTime();
-
-            } else {
-                nextStuckTime=simTime
-                        + config.getSqueezeTime();
-            }
-
-
-            if (!Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.containsKey(vehicle)){
-                Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.put(vehicle,simTime + link.getLength()/minimumRoadSpeedInMetersPerSecond);
-            } else {
-                System.out.print("");
-            }
+            gap.clear();
+        }
 
+        getInterestedInEnteringRoad().add(vehicle);
 
+        /*
+         * the first car interested in entering a road has to wait
+         * 'stuckTime' the car behind has to wait an additional stuckTime
+         * (this logic was adapted to adhere to the C++ implementation)
+         */
+        double nextStuckTime=0;
 
-            //System.out.print("enterRequest:" + link.getId() + "; vehicle:" + vehicle.getOwnerPerson().getId());
-            //System.out.println("; vehicle.getCurrentLinkId():" + vehicle.getCurrentLinkId());
+        if (getDeadlockPreventionMessages().size() > 0) {
+            nextStuckTime= getDeadlockPreventionMessages().getLast().getMessageArrivalTime() + config.getSqueezeTime();
 
-            double minTimeForNextDeadlockPreventionMessageTime=0;
+        } else {
+            nextStuckTime=simTime + config.getSqueezeTime();
+        }
 
-            if (deadlockPreventionMessages_.size() > 0) minTimeForNextDeadlockPreventionMessageTime=deadlockPreventionMessages_.getLast().getMessageArrivalTime()+0.0000000001; // ensures that deadlock prevention messages have increasing time stamps - this is assumped by original implementation around this
+        if (!Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.containsKey(vehicle)){
+            Road.getRoad(vehicle.getCurrentLinkId()).latestTimeToLeaveRoad.put(vehicle,simTime + link.getLength()/minimumRoadSpeedInMetersPerSecond);
+        } else {
+            System.out.print("");
```

Comment:
Let's get ride of  this or if prints. If some log message is required, let add proper logging

Reply:
removed this print statement